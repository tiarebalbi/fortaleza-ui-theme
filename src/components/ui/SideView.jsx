// @flow

import * as React from 'react';
import { Icon } from 'antd';
import { palette } from 'styled-theme';
import type { SideViewType } from '../../types';
import SideViewStyle from './styles/SideViewStyle';

const SideView = (props: SideViewType) => {
  const { title, component: Component, close }: SideViewType = props;
  return (
    <SideViewStyle>
      <div className="container">
        <div className="space" onClick={close} />
        <div className="content">
          <header>
            <h1>{title}</h1>
            <a onClick={close}>
              <Icon type="close-circle" />
            </a>
          </header>
          <div className="component">
            <Component {...props} />
          </div>
        </div>
      </div>
    </SideViewStyle>
  );
};

export default SideView;
