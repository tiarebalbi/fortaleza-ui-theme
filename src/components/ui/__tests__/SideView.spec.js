import React from 'react';
import { mount } from 'enzyme';
import SideView from '../../ui/SideView';

const close = jest.fn();

describe('ui/SideView', () => {
  const Component = () => <div>a</div>;
  const wrapper = mount(
    <SideView title="Custom View" close={close} component={Component} />,
  );

  beforeEach(() => close.mockClear());

  it('should be stateless', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute close function clicking on the top right button', () => {
    wrapper.find('.space').simulate('click');
    expect(close.mock.calls.length).toBe(1);
  });

  it('should execute close function clicking on blur background', () => {
    wrapper.find('header a').simulate('click');
    expect(close.mock.calls.length).toBe(1);
  });
});
