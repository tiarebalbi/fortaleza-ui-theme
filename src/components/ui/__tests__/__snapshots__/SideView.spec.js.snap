// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/SideView should render as expected 1`] = `
<SideView
  close={[MockFunction]}
  component={[Function]}
  title="Custom View"
>
  <styled.div>
    <div
      className="sc-bdVaJa fiMjLc"
    >
      <div
        className="container"
      >
        <div
          className="space"
          onClick={[MockFunction]}
        />
        <div
          className="content"
        >
          <header>
            <h1>
              Custom View
            </h1>
            <a
              onClick={[MockFunction]}
            >
              <Icon
                type="close-circle"
              >
                <i
                  className="anticon anticon-close-circle"
                />
              </Icon>
            </a>
          </header>
          <div
            className="component"
          >
            <Component
              close={[MockFunction]}
              component={[Function]}
              title="Custom View"
            >
              <div>
                a
              </div>
            </Component>
          </div>
        </div>
      </div>
    </div>
  </styled.div>
</SideView>
`;
